gravitee-am-repository/gravitee-am-repository-api/src/main/java/io/gravitee/am/repository/management/api/CommonUserRepository.java

```diff
@@ -52,4 +55,102 @@ public interface CommonUserRepository extends CrudRepository<User, String> {
     Maybe<User> findById(ReferenceType referenceType, String referenceId, String userId);
 
     Completable deleteByReference(ReferenceType referenceType, String referenceId);
+
+    class UpdateActions {
+        private final boolean profile = true;
+        private boolean role = true;
+        private boolean dynamicRole = true;
+        private boolean attributes = true;
+        private boolean entitlements = true;
+        private boolean addresses = true;
+
+        UpdateActions() {}
+
+        public boolean updateProfile() {
+            return profile;
+        }
+
+        public boolean updateRole() {
+            return role;
+        }
+
+        public UpdateActions updateRole(boolean role) {
+            this.role = role;
+            return this;
+        }
+
+        public boolean updateDynamicRole() {
+            return dynamicRole;
+        }
+
+        public UpdateActions updateDynamicRole(boolean dynamicRole) {
+            this.dynamicRole = dynamicRole;
+            return this;
+        }
+
+        public boolean updateAttributes() {
+            return attributes;
+        }
+
+        public UpdateActions updateAttributes(boolean attributes) {
+            this.attributes = attributes;
+            return this;
+        }
+
+        public boolean updateEntitlements() {
+            return entitlements;
+        }
+
+        public UpdateActions updateEntitlements(boolean entitlements) {
+            this.entitlements = entitlements;
+            return this;
+        }
+
+        public boolean updateAddresses() {
+            return addresses;
+        }
+
+        public UpdateActions updateAddresses(boolean addresses) {
+            this.addresses = addresses;
+            return this;
+        }
+
+        public boolean notOnlyProfile() {
```

Comment:
This method name is bit difficult to understand what it does, though hard to suggest a better name

Reply:
As I removed the profile boolean, I renamed this method to "updateRequire"